ts/components/countdown/CountdownProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {
+  createContext,
+  useContext,
+  useState,
+  ReactNode,
+  useRef
+} from "react";
+
+type CountdownContextType = {
+  timerCount: number;
+  resetTimer?: () => void;
+  startTimer?: () => void;
+  isRunning?: () => boolean;
+};
+
+const CountdownContext = createContext<CountdownContextType>({ timerCount: 0 });
+
+// Props type for the provider component
+interface CountdownProviderProps {
+  children: ReactNode;
+  timerTiming: number;
+}
+
+export const CountdownProvider = (props: CountdownProviderProps) => {
+  const { children, timerTiming } = props;
+  const [timerCount, setTimerCount] = useState<number>(timerTiming);
+  const interval = useRef<number | undefined>();
+
```

Comment:
```suggestion

  useEffect(
    () => () => {      
      clearInterval(interval.current);
    },
    []
  );
  
```

What do you think to put at last resort `clearInterval` on the provider unmount event? Just in case we have an edge case that we unmount with an interval that is still running?

Reply:
done al line 31! ✅